src/hpc/tls.h

```diff
@@ -55,28 +98,96 @@ typedef struct ThreadLocalStorage
   UInt LocksContended;
 } ThreadLocalStorage;
 
-#ifdef HAVE_NATIVE_TLS
+#ifdef USE_NATIVE_TLS
 extern __thread ThreadLocalStorage *TLSInstance;
 #endif
 
+#ifdef USE_PTHREAD_TLS
+
+#include <pthread.h>
+
+#ifdef ALLOW_PURE_PTHREAD_GETSPECIFIC
+// pthread_getspecific() is not defined as pure by default; redeclaring
+// it as such works for gcc/clang and allows the compiler to hoist calls
+// to pthread_getspecific() out of loops and perform constant
+// subexpression elimination on expressions containing TLS values.
+PURE_FUNC void * pthread_getspecific(pthread_key_t);
+#endif
+
+#ifdef USE_MACOS_PTHREAD_TLS_ASM
+static UInt TLSOffset;
+UInt GetTLSOffset(void);
+#else
+static pthread_key_t TLSKeyCopy;
+pthread_key_t GetTLSKey(void);
+#endif
+
+// This constructor function will initialize the static variable for
+// each C module. Making the variable static allows optimizations that
```

Comment:
Which variable is "the variable" which is being referred to here?

Reply:
Elaborated on that.